src/pages/index.js

```diff
@@ -1,106 +1,255 @@
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
-// form popup
-const newFormPopup = new PopupWithForm("#card-modal", handleFormSubmit);
-newFormPopup.setEventListeners();
+api
+  .getUserInfo()
+  .then((userData) => {
+    console.log(userData);
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setAvatar(userData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const confirmDeleteModal = new PopupWithConfirmation(
+  "#confirm-delete-modal",
+  () => {}
+);
 
-//  profile edit popup
-const userInfo = new UserInfo(".profile__title", ".profile__text");
-const profileEditPopup = new PopupWithForm("#edit-modal", (data) => {
-  userInfo.setUserInfo(data);
-  profileEditPopup.close();
-});
-profileEditPopup.setEventListeners();
+const renderCard = (card) => {
+  const cardEl = new Card(
+    {
+      data: card,
+      handleImageClick: (imgData) => {
+        cardPreviewPopup.open(imgData);
+      },
+      handleDeleteCardClick,
+      handleCardLikeClick,
+    },
+    selectors.cardTemplate
+  );
 
-// image popup
-const imagePopup = new PopupWithimage("#image-modal");
-imagePopup.setEventListeners();
+  return cardEl.getView();
+};
 
-// section class
-const section = new Section(
+const cardSection = new Section(
   {
-    items: initialCards,
     renderer: (item) => {
       const cardEl = renderCard(item);
-      section.addItem(cardEl);
+      cardSection.addItem(cardEl);
     },
   },
-  ".cards__list"
+  selectors.cardSection
+);
+
+const editPopup = new PopupWithForm("#profile-edit-modal", (obj) => {
+  handleProfileEditSubmit(obj);
+});
+
+const addPopup = new PopupWithForm("#add-card-modal", (obj) => {
+  handlePhotoAddSubmit(obj);
+});
+
+const editFormValidator = new FormValidator(
+  validationSettings,
+  profileEditForm
 );
-section.renderItems();
+const addFormValidator = new FormValidator(validationSettings, addCardModal);
 
-// functions
-function renderCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
-  return card.getView();
+const editAvatarValidator = new FormValidator(
+  validationSettings,
+  avatarEditModal
+);
+
+//Initialize all instances
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+cardPreviewPopup.setEventListeners();
+editPopup.setEventListeners();
+addPopup.setEventListeners();
+confirmDeleteModal.setEventListeners();
+editAvatarPopup.setEventListeners();
+editFormValidator.enableValidation();
+addFormValidator.enableValidation();
+editAvatarValidator.enableValidation();
+
+/*--------------------------------------------------*/
+/*                      Functions                   */
+/*--------------------------------------------------*/
+
+function handleDeleteCardClick(card) {
+  confirmDeleteModal.open();
+  confirmDeleteModal.setSubmitAction(() => {
+    const id = card.getId();
+    api
```

Comment:
Needs correction: it's also needed to notify user about loading by changing the submit button text